src/Altinn.App.Core/Internal/Events/KeyVaultSecretCodeProvider.cs

```diff
@@ -0,0 +1,43 @@
+﻿using Altinn.App.Core.Interface;
+
+namespace Altinn.App.Core.Internal.Events
+{
+    /// <summary>
+    /// Provides a secret code from Azure Key Vault to be used with
+    /// the Event system.
+    /// </summary>
+    public class KeyVaultEventSecretCodeProvider : IEventSecretCodeProvider
+    {
+        private readonly ISecrets _keyVaultClient;
+        private string _secretCode = string.Empty;
+
+        /// <summary>
+        /// Provides a new instance of <see cref="KeyVaultEventSecretCodeProvider"/>
+        /// This 
+        /// </summary>
+        /// <param name="keyVaultClient"></param>
+        public KeyVaultEventSecretCodeProvider(ISecrets keyVaultClient)
+        {
+            _keyVaultClient = keyVaultClient;
+        }
+
+        /// <inheritdoc/>
+        public async Task<string> GetSecretCode()
```

Comment:
Should we create an issue for reloading secret if it changes in keyvault without restarting the application?